### ISSUE TITLE
Add support for additional ASCII characters in named-destination

### ISSUE THREAD

**Issue by sabbyanandan** [spring-cloud/spring-cloud-dataflow#2425]
As a user, I'd like to be able to publish or consume data from named-destinations (i.e., channels) with different ASCII characters as delimiters. 

Currently, there's a strict contract (see [tokenizer.ts](https://github.com/spring-cloud/spring-cloud-dataflow-ui/blob/6d3dc92ca9bfa1383a31bd95484230229a4822e2/ui/src/app/shared/services/tokenizer.ts)) in place and only alpha-numeric, `$`, `.`, `-`,  and `_` are valid delimiters. 

However, I'd like to consume from all the topics w/ the prefix `foooz`, the data pipeline would look like:

> foo = :foooz* | filter | cassandra

Likewise, I'd like to publish data to a topic w/ wildcard regex (depends on the binders that support it) such as:

> bar = :foooz* | filter > :bar*fooz 

There are also cases for when a certain message brokers (e.g., Solace) that can have topics with `/` in it.

> baz = :fooz/baz > hdfs

In summary, it'd be nice to relax the DSL parsers (frontend & backend) to accept `*`, `/`, and `#` as valid delimiters.

**Comment by garyrussell** [comment]
>Likewise, I'd like to publish data to a topic w/ wildcard regex (depends on the binders that support it) such as:

The Kafka binder currently only supports wildcards for consumers. 

I am not even sure what a wildcard on the producer side means - distribute to all matching topics? What's the point of that? Kafka itself (and Rabbit) provides pub/sub if you want to send something to multiple apps.

**Comment by sabbyanandan** [comment]
In the DSL, when we plug a named-destination at the beginning of the stream definition, there will be a [bridge-processor](https://github.com/spring-cloud/spring-cloud-dataflow/blob/503cc905109b8d636b72c9c77ba75e272bf21fb7/spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/StreamParser.java#L149) that automatically gets deployed by SCDF. The expectation is that the processor (a consumer) will consume data from all the topics that satisfy the wildcard pattern matching. 

As for at the producer position, I see your point. An "n" number of consumers can consume from a named-destination anyways, so yeah, it doesn't seem applicable here.

Lastly, all the suggested ASCII characters in the issue depend on the binder implementation. For instance: `/` would work in solace; `*` would work in both kafka and solace etc.

**Comment by jschabowsky** [comment]
One case I could MAYBE see using wildcards on the producing side is as follows. 

A Source SCS application is using dynamic destinations to send a data to a variety of different topics based on something within the message or from the data source... Lets say its a Store and there is a topic hierarchy such that its (Using the Solace example) A/B/C where A is the StoreID, B is the transaction type and C is the Transaction Number.... If I had a destination set such that I am only store "SLC" but produce all types of Transactions and All transaction numbers this could be shown as SLC/*/* thus alerting the SCDF developer of the use of dynamic destinations and that there is a hierarchy involved. Thus you could have a flow that consumes everything just use that named destination OR create a new named destination of SLC/CASH/* to just get all transaction that are CASH from SLC. 

To summarize wildcards on the publishing side could be used to denote SCS applications that are using dynamic destinations. If this is a bad idea, how would a SCS application that uses dynamic destinations be wired?
